packages/app/src/debug/DebugContainer.cy.tsx

```diff
@@ -65,34 +59,96 @@ describe('<DebugContainer />', () => {
     })
   })
 
-  describe('render specs and tests', () => {
-    it('renders data when logged in and connected', () => {
-      const loginConnectStore = useLoginConnectStore()
+  describe('when logged in and connected', () => {
+    let loginConnectStore
+
+    beforeEach(() => {
+      loginConnectStore = useLoginConnectStore()
 
       loginConnectStore.setUserFlag('isLoggedIn', true)
       loginConnectStore.setProjectFlag('isProjectConnected', true)
+    })
+
+    it('render first pending run', () => {
       cy.mountFragment(DebugSpecsFragmentDoc, {
         onResult: (result) => {
           if (result.currentProject?.cloudProject?.__typename === 'CloudProject') {
             const test = result.currentProject.cloudProject.runByNumber
-            const other = CloudRunStubs.failingWithTests as typeof test
+            const other = CloudRunStubs.running as typeof test
+
+            other!.runNumber = 1
 
             result.currentProject.cloudProject.runByNumber = other
           }
         },
-        render: (gqlVal) => {
-          return (
-            <DebugContainer
-              gql={gqlVal}
-            />
-          )
+        render: (gqlVal) => <DebugContainer gql={gqlVal} />,
+      })
+
+      cy.findByTestId('debug-header').should('be.visible')
+      cy.findByTestId('debug-pending-splash')
+      .should('be.visible')
+      .within(() => {
+        cy.findByTestId('debug-pending-counts').should('have.text', '8 of 10 specs completed')
+      })
+    })
+
+    it('renders specs and tests when completed run available', () => {
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        onResult: (result) => {
+          if (result.currentProject?.cloudProject?.__typename === 'CloudProject') {
+            const test = result.currentProject.cloudProject.runByNumber
+            const other = CloudRunStubs.failingWithTests as typeof test
+
+            result.currentProject.cloudProject.runByNumber = other
+          }
         },
+        render: (gqlVal) => <DebugContainer gql={gqlVal} />,
       })
 
       // Only asserting that it is rendering the components for failed specs
       cy.findByTestId('debug-header').should('be.visible')
       cy.findByTestId('debug-spec-item').should('be.visible')
     })
+
+    context('newer relevant run available', () => {
+      it('displays newer run with progress when running', () => {
```

Comment:
Seeing the following for this test:
![Screen Shot 2022-12-29 at 10 02 46 AM](https://user-images.githubusercontent.com/14275198/209971623-447b4050-da24-4dac-8dd7-1d7bead40335.png)

The only reason why the pending run splash component is rendering is because it's stubbed and this is run 1, right? Otherwise we wouldn't expect to see this here.

Reply:
@astone123 Correct. This is an illogical scenario due to the stubbed "relevant run", we should not see a relevant run banner in this case once the data is wired up